test/utils/Arrays.test.js

```diff
@@ -154,22 +116,79 @@ describe('Arrays', function () {
     }
   });
 
-  describe('unsafeAccess', function () {
-    for (const [type, { artifact, elements }] of Object.entries({
-      address: { artifact: 'AddressArraysMock', elements: randomArray(generators.address, 10) },
-      bytes32: { artifact: 'Bytes32ArraysMock', elements: randomArray(generators.bytes32, 10) },
-      uint256: { artifact: 'Uint256ArraysMock', elements: randomArray(generators.uint256, 10) },
-    })) {
-      describe(type, function () {
-        describe('storage', function () {
-          const fixture = async () => {
-            return { instance: await ethers.deployContract(artifact, [elements]) };
-          };
+  for (const [type, { artifact, elements, comp }] of Object.entries({
+    address: {
+      artifact: 'AddressArraysMock',
+      elements: Array.from({ length: 10 }, generators.address),
+      comp: (a, b) => bigintSign(ethers.toBigInt(a) - ethers.toBigInt(b)),
+    },
+    bytes32: {
+      artifact: 'Bytes32ArraysMock',
+      elements: Array.from({ length: 10 }, generators.bytes32),
+      comp: (a, b) => bigintSign(ethers.toBigInt(a) - ethers.toBigInt(b)),
+    },
+    uint256: {
+      artifact: 'Uint256ArraysMock',
+      elements: Array.from({ length: 10 }, generators.uint256),
+      comp: (a, b) => bigintSign(a - b),
+    },
+  })) {
+    describe(type, function () {
+      const fixture = async () => {
+        return { instance: await ethers.deployContract(artifact, [elements]) };
+      };
+
+      beforeEach(async function () {
+        Object.assign(this, await loadFixture(fixture));
+      });
+
+      describe('sort', function () {
+        for (const length of [0, 1, 2, 8, 32, 128]) {
+          describe(`${type}[] of length ${length}`, function () {
+            beforeEach(async function () {
+              this.elements = Array.from({ length }, generators[type]);
+            });
+
+            it('sort array', async function () {
+              // nothing to do here, beforeEach and afterEach already take care of everything.
+            });
 
-          beforeEach(async function () {
-            Object.assign(this, await loadFixture(fixture));
+            if (length > 1) {
```

Comment:
what happens if `length <= 1`? We should test that case too

Reply:
if length is 0 or 1, then any random array is already: shuffled, sorted, and reverse sorted (simultaneously).
No need to check.

Reply:
If length is 0 or 1, we do the "normal" test ... and that is it ... not need for particaular cases

Reply:
Right. The logic in the `afterEach is a bit disconcertant`.

I'd suggest having it below the `beforeEach` for clarity. But not a blocker.